circle.yml

```diff
@@ -137,14 +137,24 @@ defaults:
       command: |
         ~/build/evmc/test/evmc-vmtester ~/build/src/libhera.so
 
-  evm2wasm-test: &evm2wasm-test
+  evm2wasm-direct-test: &evm2wasm-direct-test
     run:
-      name: "Run evm2wasm state tests"
+      name: "Run evm2wasm state tests (using evm2wasm.cpp directly)"
       command: |
         testeth --version
         testeth -t GeneralStateTests/stExample -- --testpath tests --vm ~/build/src/libhera.so --singlenet Byzantium --singletest "add11" --evmc evm1mode=evm2wasm.cpp
         testeth -t GeneralStateTests/stStackTests -- --testpath tests --vm ~/build/src/libhera.so --singlenet Byzantium --evmc evm1mode=evm2wasm.cpp
 
+  evm2wasm-contract-test: &evm2wasm-contract-test
+    run:
+      name: "Run evm2wasm state tests (using runevm contract)"
+      command: |
+        curl -o /tmp/runevm.wasm -L https://github.com/axic/runevm/releases/download/prototype/runevm.wasm
+        #test "$(shasum /tmp/runevm.wasm)" = "598f87f072f66e2269dd6919292b2934dbb20492  /tmp/runevm.wasm"
+        testeth --version
+        testeth -t GeneralStateTests/stExample -- --testpath tests --vm ~/build/src/libhera.so --singlenet Byzantium --singletest "add11" --evmc evm1mode=evm2wasm --evmc sys:evm2wasm=/tmp/runevm.wasm
```

Comment:
I am pretty sure that you do not need to use `--evmc` preceding each option. Last I checked (possibly a previous version of EVMC), it worked fine with one `--evmc` invocation for all options.

Reply:
How would that work? `--evmc "evm1mode=evm2wasm sys:evm2wasm=/tmp/runevm.wasm"` ?

I do think it needs separate ones because that's how parsing is done by aleth.

cc @chfast

Reply:
Checking...

Reply:
Now you have to do `--evmc a=b --evmc x=y`, but I'm working on a change to allow `--evmc a=b x=y`.

Reply:
Does it make sense to have the latter supported?

Reply:
What's the "latter"?

Reply:
`--evmc a=b x=y`

Reply:
It's shorted isn't it?

Reply:
What happens if the value part of the option has spaces?

Reply:
`--evmc "a=b and c" x=y`?

Reply:
I find it confusing when parameters can have spaces, e.g. if there are multiple EVMC options in one line I'd do `--evmc "a=b x=y"` and in taht case `b` or `y` cannot have spaces

Reply:
Maybe, but I'm not going to write a custom option parser just for EVMC options. Now it is simple as finding first `=` and splitting the string accordingly.

If you want you can check whenever CLI11 library offers any fancy option grouping we could use.

Reply:
Well I guess the simple solution is not changing how it works currently? :)

Reply:
It's not a change, it's an extension.

Reply:
Added in https://github.com/ethereum/aleth/pull/5234.

Okay, if the old way is still accepted then I have nothing against it.